samples/resourceops/volume_snapshotop_rokurl.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import kfp.dsl as dsl
+
+
+@dsl.pipeline(
+    name="VolumeSnapshotOp RokURL",
+    description="The fifth example of the design doc."
+)
+def volume_snapshotop_rokurl(rok_url):
+    vop1 = dsl.VolumeOp(
+        name="create_volume_1",
+        resource_name="vol1",
+        size="1Gi",
+        annotations={"rok/origin": rok_url},
+        modes=dsl.VOLUME_MODE_RWM
+    )
+
+    step1 = dsl.ContainerOp(
+        name="step1_concat",
+        image="library/bash:4.4.23",
+        command=["sh", "-c"],
+        arguments=["cat /data/file*| gzip -c >/data/full.gz"],
+        volumes={"/data": vop1.volume}
+    )
+
+    step1_snap = dsl.VolumeSnapshotOp(
+        name="create_snapshot_1",
+        resource_name="snap1",
+        volume=step1.volume
+    )
+
+    vop2 = dsl.VolumeOp(
+        name="create_volume_2",
+        resource_name="vol2",
+        data_source=step1_snap.snapshot,
+        size=step1_snap.outputs["size"]
+    )
+
+    step2 = dsl.ContainerOp(
+        name="step2_gunzip",
+        image="library/bash:4.4.23",
+        command=["gunzip", "-k", "/data/full.gz"],
+        volumes={"/data": vop2.volume}
+    )
+
+    step2_snap = dsl.VolumeSnapshotOp(
+        name="create_snapshot_2",
+        resource_name="snap2",
+        volume=step2.volume
```

Comment:
Does this sample focus on having one volume per output? Could we cover the case where the same volume is reused throughout the workflow, but a different directory within the volume is used for each output?

Reply:
@vicaire 
> Does this sample focus on having one volume per output? Could we cover the case where the same volume is reused throughout the workflow, but a different directory within the volume is used for each output?

We have modified the `resourceops/volume_snapshotop_sequential.py` sample, which covers exactly your desired case: One volume, being used by all steps, each step storing output in a different directory.